magenta-lib/src/test/scala/magenta/input/DeploymentTypeResolverTest.scala

```diff
@@ -0,0 +1,55 @@
+package magenta.input
+
+import magenta.fixtures._
+import org.scalatest.{EitherValues, FlatSpec, Matchers}
+
+class DeploymentTypeResolverTest extends FlatSpec with Matchers with EitherValues {
+  val deployment = Deployment("bob", "stub-package-type", List("stack"), List("eu-west-1"), actions=None, "bob", "bob", Nil, Map.empty)
+  val deploymentTypes = List(stubPackageType(Seq("upload", "deploy")))
+
+  "validateDeploymentType" should "fail on invalid deployment type" in {
+    val deploymentWithInvalidType = deployment.copy(`type` = "invalidType")
+    val configError = DeploymentTypeResolver.validateDeploymentType(deploymentWithInvalidType, deploymentTypes).left.value
+    configError.context shouldBe "bob"
+    configError.message should include(s"Unknown type invalidType")
+  }
+
+  it should "fail if explicitly given empty actions" in {
+    val deploymentWithNoActions = deployment.copy(actions = Some(Nil))
+    val configError = DeploymentTypeResolver.validateDeploymentType(deploymentWithNoActions, deploymentTypes).left.value
+    configError.context shouldBe "bob"
+    configError.message should include(s"Either specify at least one action or omit the actions parameter")
```

Comment:
Some of these are just checking the exact error message, maybe easier to just match directly (or do `have`) in those cases?